### PR TITLE
Security Annotations

### PR DESCRIPTION
**Pull request by tejusredlapalle**

### CODE REVIEW

File: basex-api/src/main/java/org/basex/http/HTTPContext.java [hunk 1]
```diff
@@ -106,6 +106,7 @@
    */
   void authorize() throws BaseXException {
     final String value = req.getHeader(AUTHORIZATION);
+    System.out.println(value);
```
Comment (LeoWoerteler) [inline basex-api/src/main/java/org/basex/http/HTTPContext.java]: `sysout`s should probably be removed.

File: basex-api/src/main/java/org/basex/http/HTTPContext.java [hunk 2]
```diff
@@ -311,6 +312,7 @@ public Context authenticate() throws IOException {
           sresponse.append(':' + map.get(NC) + ':' + map.get(CNONCE) + ':' + qop);
         }
         sresponse.append(':' + ha2);
+        System.out.println(Strings.md5(sresponse.toString())+ ' '+password);
```
Comment (LeoWoerteler) [inline basex-api/src/main/java/org/basex/http/HTTPContext.java]: Same as above.

File: basex-core/src/main/java/org/basex/query/func/http/HttpClient.java [hunk 3]
```diff
@@ -161,17 +161,75 @@ private static void setContentType(final HttpURLConnection conn, final HttpReque
    * @param conn HTTP connection
    * @param r request data
    * @throws QueryException query exception
+   * @throws IOException
    */
-  private void setRequestHeaders(final HttpURLConnection conn, final HttpRequest r)
-      throws QueryException {
+  private void setRequestHeaders(HttpURLConnection conn, final HttpRequest r)
```
Comment (LeoWoerteler) [inline basex-core/src/main/java/org/basex/query/func/http/HttpClient.java]: `final` keyword is missing, did you activate Checkstyle?

File: basex-core/src/main/java/org/basex/query/func/http/HttpClient.java [hunk 4]
```diff
@@ -161,17 +161,75 @@ private static void setContentType(final HttpURLConnection conn, final HttpReque
...
+      throws QueryException, IOException {
 
     for(final byte[] headers : r.headers)
       conn.addRequestProperty(string(headers), string(r.headers.get(headers)));
-    // HTTP Basic Authentication
+
     final byte[] sendAuth = r.attrs.get(SEND_AUTHORIZATION);
-    if(sendAuth != null && Bln.parse(sendAuth, info))
+    final byte[] am = r.attrs.get(AUTH_METHOD);
+    System.out.println(string(am));
```
Comment (LeoWoerteler) [inline basex-core/src/main/java/org/basex/query/func/http/HttpClient.java]: Should be removed.

File: basex-core/src/main/java/org/basex/query/func/http/HttpClient.java [hunk 5]
```diff
@@ -161,17 +161,75 @@ private static void setContentType(final HttpURLConnection conn, final HttpReque
...
+      if(sendAuth != null && Bln.parse(sendAuth, info))
       conn.setRequestProperty(AUTHORIZATION,
       encodeCredentials(string(r.attrs.get(USERNAME)), string(r.attrs.get(PASSWORD))));
+  } else if(!string(am).equals(null) && string(am).equals(DIGEST)) {
+   // HTTP Digest Authentication
+      if(sendAuth != null && Bln.parse(sendAuth, info))
+      //try {
+        conn.setRequestProperty(AUTHORIZATION, DIGEST);
+        String sr = conn.getHeaderField(WWW_AUTHENTICATE);
+        System.out.println(sr);
```
Comment (LeoWoerteler) [inline basex-core/src/main/java/org/basex/query/func/http/HttpClient.java]: remove

File: basex-core/src/main/java/org/basex/query/func/http/HttpClient.java [hunk 6]
```diff
@@ -161,17 +161,75 @@ private static void setContentType(final HttpURLConnection conn, final HttpReque
...
+        String ha2 = Strings.md5("GET:" + uri);
+        String response = Strings.md5(ha1 + ":" + nonce + ":" + nc + ":" + cnonce + ":" + qop + ":" + ha2);
+
+        String creds = "username=" + username + ", " + "realm=" + realm + ", " + "nonce=" + nonce
+            + ", " + "uri=" + uri + ", " + "qop=" + qop + ", " + "nc=" + nc + ", " + "cnonce=" + cnonce
+            + ", " + "response=" + response;
+
+        conn = openConnection(string(r.attrs.get(HREF)));
+        conn.setAllowUserInteraction(true);
+        System.out.println(creds);
```
Comment (LeoWoerteler) [inline basex-core/src/main/java/org/basex/query/func/http/HttpClient.java]: remove

File: basex-core/src/main/java/org/basex/query/func/http/HttpClient.java [hunk 7]
```diff
@@ -161,17 +161,75 @@ private static void setContentType(final HttpURLConnection conn, final HttpReque
...
+        conn.setRequestProperty(AUTHORIZATION, DIGEST + ' ' + creds);
+  } else {
+    System.out.println("Specify Authentication");
```
Comment (LeoWoerteler) [inline basex-core/src/main/java/org/basex/query/func/http/HttpClient.java]: Proper error handling?

**Comment by tejusredlapalle** [comment]
@LeoWoerteler : The "sysouts" were basically for debugging and not production.
My mistake, pull request was opened accidentally :(

**Comment by LeoWoerteler** [comment]
No worries, it happens to [other people](https://github.com/BaseXdb/basex/commit/08f5752) as well ;-). This was just normal code review, I am excited to see contributions from someone else than @ChristianGruen and myself in the core project!

**Comment by tejusredlapalle** [comment]
Yes, development and debugging done in parallel :+1: But, thanks for looking into it :-) !

**Pull request closed without merging**